File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, bool> _map;
        for(int i=0; i<nums.size(); i++){
            if(_map.contains(nums[i])){
                return true;
            }
            _map[nums[i]] = true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set()
        for num in nums:
            seen.add(num)
        longest_sequence = 0
        for i, num in enumerate(nums):
            if num-1 not in seen:
                #start of a sequence
                curr_seq = 0

                while num in seen:
                    curr_seq +=1
                    num+=1
                longest_sequence = max(longest_sequence, curr_seq )
        return longest_sequence
